sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs

```diff
@@ -93,6 +93,21 @@ private async ValueTask<bool> AuthorizeRequestOnChallengeAsyncInternal(HttpMessa
 
             string authority = GetRequestAuthority(message.Request);
             string scope = AuthorizationChallengeParser.GetChallengeParameterFromResponse(message.Response, "Bearer", "resource");
+
+            string error = AuthorizationChallengeParser.GetChallengeParameterFromResponse(message.Response, "Bearer", "error");
```

Comment:
Do we have tests for these new logic branches?

Reply:
I just added 3 tests for this:
- `GetClaimsFromChallengeHeaders` to make sure we're getting and decoding the claims correctly,
- `HandlesCaeChallenges` to make sure it can process a CAE challenge after a scope challenge,
- `ThrowsWithTwoConsecutiveCaeChallenges` to ensure that the code won't loop forever when it gets one CAE challenge after another (although, this is a very unlikely scenario).